### PR TITLE
Add support for applicationAccessEnabled

### PR DESCRIPTION
**Pull request by VivekVinushanth**
**Related Issue**
- https://github.com/wso2/product-is/issues/20378

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

**Comment by VivekVinushanth** [comment]
DO NOT MERGE before:
- https://github.com/wso2/carbon-identity-framework/pull/5701

**Review by AnuradhaSK — DISMISSED** [review]

File: components/org.wso2.carbon.identity.api.server.application.management/org.wso2.carbon.identity.api.server.application.management.v1/src/main/resources/applications.yaml [hunk 1]
```diff
@@ -3061,6 +3061,10 @@ components:
           description: Decides whether the application is visible for end users.
         certificate:
           $ref: '#/components/schemas/Certificate'
+        applicationAccessEnabled:
+          type: boolean
+          example: true
+          description: Decides whether the application can be accessed.
```
Comment (AnuradhaSK) [inline components/org.wso2.carbon.identity.api.server.application.management/org.wso2.carbon.identity.api.server.application.management.v1/src/main/resources/applications.yaml:3067]: Better to treat this property as a primary property as similar to  isManagementApp 
 https://github.com/wso2/identity-api-server/pull/612/files#diff-37b5c0ba2971cd297cefdc03e49736637886f855740d6071634220bdf1f15615R2921

Comment (VivekVinushanth) [reply components/org.wso2.carbon.identity.api.server.application.management/org.wso2.carbon.identity.api.server.application.management.v1/src/main/resources/applications.yaml]: Addressed with https://github.com/wso2/identity-api-server/pull/612/commits/72322e078e04a1e5a6bb598c202f047c4a807070

File: components/org.wso2.carbon.identity.api.server.application.management/org.wso2.carbon.identity.api.server.application.management.v1/src/main/java/org/wso2/carbon/identity/api/server/application/management/v1/core/functions/application/ServiceProviderToApiModel.java [hunk 2]
```diff
@@ -102,6 +102,7 @@ public class ServiceProviderToApiModel implements Function<ServiceProvider, Appl
     private static final Set<String> systemApplications = ApplicationManagementServiceHolder
             .getApplicationManagementService().getSystemApplications();
     private static final String IS_FRAGMENT_APP = "isFragmentApp";
+    private static final String APPLICATION_ACCESS_ENABLED = "IsApplicationAccessEnabled";
```
Comment (AnuradhaSK) [inline components/org.wso2.carbon.identity.api.server.application.management/org.wso2.carbon.identity.api.server.application.management.v1/src/main/java/org/wso2/carbon/identity/api/server/application/management/v1/core/functions/application/ServiceProviderToApiModel.java:105]: Better if we can have the similar name in API and backend service. `IsApplicationEnabled` WDYT?

Comment (VivekVinushanth) [reply components/org.wso2.carbon.identity.api.server.application.management/org.wso2.carbon.identity.api.server.application.management.v1/src/main/java/org/wso2/carbon/identity/api/server/application/management/v1/core/functions/application/ServiceProviderToApiModel.java:105]: Need to change the Framework then.

Comment (VivekVinushanth) [reply components/org.wso2.carbon.identity.api.server.application.management/org.wso2.carbon.identity.api.server.application.management.v1/src/main/java/org/wso2/carbon/identity/api/server/application/management/v1/core/functions/application/ServiceProviderToApiModel.java:105]: Merged the Integration test PR with this name as well.

**Comment by VivekVinushanth** [comment]
Will not trigger PR builder for this change.

**Review by AnuradhaSK — APPROVED** [review]

**Pull request merged**